### PR TITLE
BR-102 Revert skipIT changes to DDF, and use global maven-settings

### PR DESCRIPTION
**Pull request by TonyMorrison**
#### What does this PR do?
Make skipITs default by adding it to the configuration for the plugin in the pom
Added the property for skipITs to the cx-internal-ci profile in the global maven settings file
added -DskipITs=false for the Integration Test stage in the Jenkinsfile 

#### Who is reviewing it? 
@shaundmorris 
@bdeining 
@LinkMJB 

#### Select relevant component teams: 
@codice/build 
@codice/continuous-integration 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@shaundmorris

#### How should this be tested?
Build should complete successfully, no ITests in the stage for no Itests, ITests in the ITest Only stage
#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -65,11 +65,9 @@ pipeline {
                 timeout(time: 3, unit: 'HOURS')
             }
             steps {
-                // TODO: Maven downgraded to work around a linux build issue. Falling back to system java to work around a linux build issue. re-investigate upgrading later
-                // -DskipITs is temporary to skip all the tests that were failing at the time. See https://github.com/codice/ddf/issues/5777
                 withMaven(maven: 'maven-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
                     sh 'mvn install -B -DskipStatic=true -DskipTests=true $DISABLE_DOWNLOAD_PROGRESS_OPTS'
-                    sh 'mvn clean install -B -DskipITs -pl !$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+                    sh 'mvn clean install -B -Pcx-internal-ci -pl !$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET $DISABLE_DOWNLOAD_PROGRESS_OPTS'
```
Comment (LinkMJB) [inline Jenkinsfile:70]: @TonyMorrison can you help connect the dots for me, on what the `cx-internal-ci` profile accomplishes? That one is new to me 😄

File: Jenkinsfile [hunk 2]
```diff
@@ -96,11 +92,10 @@ pipeline {
                 timeout(time: 3, unit: 'HOURS')
             }
             steps {
-                // TODO: Maven downgraded to work around a linux build issue. Falling back to system java to work around a linux build issue. re-investigate upgrading later
                 withMaven(maven: 'maven-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
                     sh '''
                         unset JAVA_TOOL_OPTIONS
-                        mvn install -B -pl $ITESTS -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
+                        mvn install -B -DskipITs=false -pl $ITESTS -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS
```
Comment (bdeining) [inline Jenkinsfile:98]: does this explicitly need to be set to false?  or can it just be removed?

File: Jenkinsfile (hunk 1 above)
Comment (TonyMorrison) [reply Jenkinsfile:70]: it is a profile in the global-maven-settings in which i put a property, skipITs set to true. In the pom i added a new configuration for skipITs with the value coming from this property. This makes skipITs true by default if you use the cx-internal-ci. 
For what I am doing, this makes (theoretical, unless I am accidentally getting the results via something else) skipITs true for stages where I wanted it and in the one stage where we wanted the ITests we just set the property to false.

File: Jenkinsfile (hunk 2 above)
Comment (TonyMorrison) [reply Jenkinsfile:98]: i can test this, it may work if we dont set a value for this

**Comment by LinkMJB** [comment]
build now

File: Jenkinsfile (hunk 1 above)
Comment (shaundmorris) [reply Jenkinsfile:70]: Would it make more sense to call the profile `skipITs`? Is there anything else in the profile?

**Comment by AdamBurstynski** [comment]
I'm not sure this is a good idea.
-DskipITs is parameter provided by the failsafe plugin. Using it directly here means anyone familiar with Maven will know what we are doing. Moving it into a profile means we lose locality and requires more tribal knowledge to understand what the build is doing.

Another thing to keep in mind the flag to turn off the profile parameter needs to be merged into every branch. You can already see this by looking at the master builds, the "ITests" are finishing in seconds and they have been since at least the 14th of the month.

**Comment by TonyMorrison** [comment]
Closing this PR no change currently necessary

**Pull request closed without merging**